implementation/src/main/java/io/smallrye/config/ConfigMapping.java

```diff
@@ -40,4 +40,23 @@
      * @return the configuration property prefix
      */
     String prefix() default "";
+
+    /**
+     * The naming strategy to use for the config mapping. This only matters for method names that contain both
+     * lower case and upper case characters.
+     *
+     * @return the config mapping naming strategy.
+     */
+    NamingStrategy namingStrategy() default NamingStrategy.KEBAB_CASE;
+
+    enum NamingStrategy {
```

Comment:
Hi, I think this should be expandable with custom naming strategies. We are using underscores for instance.

Reply:
You mean you provide your own custom strategy? Or just add one that supports underscore?

Reply:
I meant it would be nice if the code would allow for custom providers (e.g. via SPI?).
However, a default UNDERSCORE variant would be nice too ;).

Reply:
Ok, let me see what I can do :)